src/Microsoft.AspNetCore.Rewrite/Extensions/HttpRequestExtensions.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Rewrite.Extensions
+{
+    public static class HttpRequestExtensions
+    {
+        public static Uri ToUri(this HttpRequest request)
```

Comment:
cool, didn't know about `UriHelper`. i changed the implementation to use it. @Tratcher are you saying get rid of the extension method altogether? it's a lot of set up at each call site if so.